ballista/rust/scheduler/src/state/mod.rs

```diff
@@ -279,8 +291,20 @@ impl SchedulerState {
     pub async fn assign_next_schedulable_task(
         &self,
         executor_id: &str,
+        job_ids: Vec<String>,
     ) -> Result<Option<(TaskStatus, Arc<dyn ExecutionPlan>)>> {
-        let tasks = self.get_all_tasks().await?;
+        // Will assign the task with same jobId to same executor to improve locality.
+        let tasks;
+        let mut tasks_by_job_id: HashMap<String, TaskStatus> = Default::default();
+        for job_id in job_ids {
+            tasks_by_job_id.extend(self.get_tasks_with_job_id(&job_id).await?);
+        }
+        tasks_by_job_id.retain(|_, v| v.status.is_none());
```

Comment:
i think this filter could have been done earlier in `get_tasks_with_job_id` so we won't need to go through the hashmap twice.